File: python/46.py
from math import sqrt


def is_prime_basic(x):
    if x <= 1:
        return False
    elif x <= 3:
        return True
    elif x % 2 == 0 or x % 3 == 0:
        return False
    i = 5
    while i ** 2 <= x:
        if x % i == 0 or x % (i+2) == 0:
            return False
        i += 6
    return True

primes = [2]

i = 3
while True:
    print(i, "\r", end="")
    if is_prime_basic(i):
        primes.append(i)
    else:
        for x in primes:
            t = sqrt((i - x) / 2)
            dr = False
            if int(t) == t:
                dr = True
                break
        if not dr:
            print(i)
            break
    i += 2

File: cpp/46.cpp
#include <iostream>
#include <cmath>
#include <vector>

using namespace std;


bool is_prime(int n){
    if (n <= 1)
    {
        return false;
    } else if (n <= 3)
    {
        return true;
    } else if (n % 2 == 0 || n % 3 == 0)
    {
        return false;
    }

    int i =5;

    while (i * i <= n)
    {
        if (n % i == 0 || n % (i+2) == 0)
        {
            return false;
        }
        i += 6;
    }
    return true;
}


int main() {
    int i {3};
    float t{};
    bool dr {};
    vector<int> primes {2};


    while (true)
    {
        if (is_prime(i)){
            primes.push_back(i);
        } else {
            for (int x: primes)
            {
                t = sqrt((i -x) / 2);
                dr = false;
                if (ceil(t) == t){
                    dr =true;
                    break;
                }
            }

            if (not dr)
                break;
        }
        i += 2;
    }

    cout << i << endl;


    return 0;
}